### DISCUSSION TITLE
Update check of main react native libraries

### DISCUSSION THREAD

**Discussion by lfoliveir4** [microsoft/rnx-kit#1622] [Ideas] [answered]
Currently react native offers a large number of libraries that can be used by engineers. In this context, rnx-kit proposes to be a powerful library management tool. It is notable that rnx-kit is a new and still limited tool, but the big discussion I want to bring up here is:

What is the best way for rnx-kit to include updating the main libraries used by react native engineers?

Below are some libraries that would be examples:

@react-navigation/material-top-tabs
@react-navigation/drawer
react-native-camera
lottie-react-native
react-native-image-picker
react-native-maps
react-native-vector-icons
@react-native-community/masked-view

**Comment by kelset** [comment, marked as answer]
The answer to this is somewhat complementary to [my comment here](https://github.com/microsoft/rnx-kit/issues/1621).
Basically, there are 2 approaches:
1. you can make a [local custom profile](https://microsoft.github.io/rnx-kit/docs/guides/dependency-management#customization) that you can use to enhance the default list of capabilities that [rnx-kit comes with](https://github.com/microsoft/rnx-kit/tree/main/packages/dep-check#capabilities).
2. the other part is a more "sistemic" approach, and it's something that we have started to investigate here: https://github.com/microsoft/rnx-kit/issues/1550 and around which @tido64 has already started to do some work (https://github.com/microsoft/rnx-kit/pull/1580) - basically, we want to have multiple "default profiles" that you can pick from directly from dep-check. Subscribe to that issue to stay in the loop of when it's done :)

**Reply by lfoliveir4** [reply]
@kelset Thanks for answering!
